api/src/routes/crud.ts

```diff
@@ -501,21 +501,91 @@ router.post("/medication-request", [requireJWTMiddleware], async (req: Request,
                 }
             
             } else {
-                console.log(`Error: property ${medication} not found`);                
+                console.log(`Error: property ${medication} not found`);
             }
 
             const { siteCoding, routeCoding, methodCoding, unitCoding } = administrationCodings;
-            const medicationRequestDetails = MedicationRequest({ id, ...req.body[medication] }, medicationCoding, siteCoding, routeCoding, methodCoding, unitCoding);    
+            const medicationRequestDetails = MedicationRequest({ id, ...req.body[medication] }, medicationCoding, siteCoding, routeCoding, methodCoding, unitCoding);
             
-            await FhirApi({ url: `/MedicationRequest/${id}`, method: "PUT", data: JSON.stringify(medicationRequestDetails) });              
+            await FhirApi({ url: `/MedicationRequest/${id}`, method: "PUT", data: JSON.stringify(medicationRequestDetails) });
             
             medicationRequestIds.push({ medication, status: "success", medicationRequestId: id });
-        }     
+        }
         
-        res.json({ status: "success", data: medicationRequestIds });                    
+        res.json({ status: "success", data: medicationRequestIds });
     } catch (err) {
         res.json({ err, status: "error" });
     }
-})
+});
+
+router.post("/allergy-intolerance", [requireJWTMiddleware], async (req: Request, res: Response) => {
+    
+    try {
+
+        let id = uuidv4();
+
+        const { patientId, vaccine } = req.body;
```

Comment:
Currently, we are not validating the patient id. From the client it is submitted from a variable stored in the local storage as it is only created once during patient registration. Regarding if its missing, if you want, I can include a statement to check its availability. If the patient id is invalid, the HAPI FHIR server will reject it and return an appropriate error informing the same since it will check if the provided id exists in the patients resources.